src/rgw/rgw_rest_log.cc

```diff
@@ -815,6 +815,130 @@ void RGWOp_MDLog_Status::send_response()
   flusher.flush();
 }
 
+class RGWOp_MDLog_CompareStatus : public RGWRESTOp {
+  rgw_meta_sync_status status;
+  RGWMetadataLogInfo info;
+  list<cls_log_entry> entries;
+  string last_marker;
+  bool truncated;
+  ceph::real_time *current_timestamp, *latest_timestamp;
+public:
+  RGWOp_MDLog_CompareStatus() : truncated(false) {}
+  ~RGWOp_MDLog_CompareStatus() override {}
+
+  int check_caps(const RGWUserCaps& caps) override {
+    return caps.check_cap("mdlog", RGW_CAP_READ);
+  }
+  int verify_permission() override {
+    return check_caps(s->user->get_caps());
+  }
+  void execute() override;
+  void send_response() override;
+  const char* name() const override {
+    return "compare_timestamps";
+  }
+};
+
+void RGWOp_MDLog_CompareStatus::execute() {
+  for(int i=0;i<64;i++) {
```

Comment:
64 is hardcoded?

Reply:
I think for metadata, this api is getting 64 markers as input. So, It is iterating over it for 64 times